DataMigration/src/main/java/org/sagebionetoworks/dian/datamigration/SynapseUtil.java

```
@@ -0,0 +1,970 @@
+        for (String filePath : filePathList) {
+            File file = new File(filePath);
+            migrationFileCount++;
+            ObjectMapper mapper = new ObjectMapper();
+            InputStream is = new FileInputStream(file);
+            ParticipantScheduleData obj = mapper.readValue(is, ParticipantScheduleData.class);
+
+            if (obj == null || obj.participant_id == null) {
+                failedToParseList.add(filePath);
+            } else {
+                String key = obj.participant_id;
+                File existingFile = map.get(key);
+                if (existingFile == null) {
+                    map.put(key, file);
+                } else {
+                    // There are multiple schedule entries per user,
+                    // So we want to get the most recent one by looking at the filename
+                    // which will always end with an iso 8601 date and ".json"
+                    String existingFilePath = existingFile.getAbsolutePath();
+                    if (existingFilePath.length() >= filenameDateSize) {
+                        existingFilePath = existingFilePath.substring(
+                                existingFilePath.length() - filenameDateSize);
+                    }
+                    String newFilePath = file.getAbsolutePath();
+                    if (newFilePath.length() >= filenameDateSize) {
+                        newFilePath = newFilePath.substring(
+                                newFilePath.length() - filenameDateSize);
+                    }
+                    if (newFilePath.compareTo(existingFilePath) > 0) {
+                        map.put(key, file);
+                    }
+                }
+            }
+        }
+
+        System.out.println("Test sessions parsing complete\nParsed " + migrationFileCount + " files.");
+
+        if (!failedToParseList.isEmpty()) {
+            System.out.println("Failed to parse file(s) " +
```

What do the failures mean?  Do they indicate that manual intervention is needed?  Should we throw an exception and stop the pipeline?